eslint-plugin-prettier.js

```diff
@@ -224,4 +216,30 @@ const eslintPluginPrettier = {
   },
 };
 
+// Assign configs after initial plugin configuration, as flat configs
+// require referencing the original plugin that contains the rules.
+Object.assign(eslintPluginPrettier.configs, {
+  recommended: {
+    extends: 'prettier',
+    plugins: ['prettier'],
+    rules: {
+      'prettier/prettier': 'error',
+      'arrow-body-style': 'off',
+      'prefer-arrow-callback': 'off',
+    },
+  },
+  ['recommended-flat']: {
```

Comment:
Should we use camelCase here?

Reply:
looks like config names should be as rules with dashes also eslint have config naming convention for eslint plugins with dashes too

Reply:
The previous rules are for legacy configs, for flat configs, we're using js object reference instead, so camelCase is preferred IMO.

Reply:
`recommended-flat` is all based on direction from the ESLint team:

> Export both eslintrc and flat configs. The configs key is only validated when a config is used, so you can provide both formats of configs in the configs key. We recommend that you append older format configs with -legacy to make it clear that these configs will not be supported in the future. For example, if your primary config is called recommended and is in flat config format, then you can also have a config named recommended-legacy that is the eslintrc config format.

https://eslint.org/docs/latest/extend/plugin-migration-flat-config#backwards-compatibility

They're recommending appending `-legacy` for legacy configs, which means `-flat` is the equivalent. Should we not go for that?

Reply:
That strategy is an unnecessary breaking change IMO, and it can be made in next major version.

Reply:
Yes, that's why the suggestion is to add `recommended-flat` now (non-breaking) and swap them around in next major.

Reply:
My suggestion is to add `flatRecommended` config key instead of `recommended-flat`.